### PR TITLE
Closes #145 documentation for datasets with xxTEST and xxTESTCD

### LINKED ISSUE #145: Documentation:  For datasets with `xxTEST`/`xxTESTCD` variables, list out distinct values in documentation
We currently have a nice documentation pipeline to create reference pages for each SDTM dataset we provide. It would be nice to detect if the dataset is "BDS" structure and thus contains  `xxTEST`/`xxTESTCD` variables. If it does, we could list out distinct values in he documentation by adding an additional section. For instance, for `VS` it would look something like this (but in a table):

## Available test codes

This dataset contains the following `VSTEST`/`VSTESTCD` combinations:

- Diastolic Blood Pressure - DIABP   
- Height - HEIGHT  
- Pulse Rate  -  PULSE   
- Systolic Blood Pressure  - SYSBP   
- Temperature - TEMP    
- Weight - WEIGHT  
- 
Note - this issue is similar to the [corresponding one in pharmaverseadam](https://github.com/pharmaverse/pharmaverseadam/issues/101)

### PR DESCRIPTION
**Pull request by Lina2689**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by Fanny-Gautier — CHANGES_REQUESTED** [review]
@manciniedoardo , could you please take a deeper look? Both Lina and I worked on the code, and we’d appreciate a fresh perspective. Thank you.

It's already looking very good to me! Thanks Lina 👏

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
One or two minor points, but it looks amazing, well done! 😍 I was also wondering if the Key features section of the script should also include a line talking about this new functionality?

File: data-raw/create_sdtms_data.R [hunk 1]
```diff
@@ -109,20 +115,45 @@ for (dataset_name in datasets) {
   dataset <- get(dataset_name)
   metadata <- specs[specs$name == dataset_name, ]
 
+  # Identify column names ending in TEST and TESTCD
+  test_col <- names(dataset)[grepl("TEST$", names(dataset))]
+  testcd_col <- names(dataset)[grepl("TESTCD$", names(dataset))]
+
+  if (length(test_col) == 1 && length(testcd_col) == 1) {
+    # Check both columns exist
+    unique_tests <- unique(dataset[c(testcd_col, test_col)])
```
Comment (bundfussr) [inline data-raw/create_sdtms_data.R:124]: I would sort the test codes alphabetically.

File: data-raw/create_sdtms_data.R [hunk 2]
```diff
@@ -109,20 +115,45 @@ for (dataset_name in datasets) {
...
+
+    # Start building the documentation as a character vector
+    # Add each testcd/test pair as an \item{}
+    testnames <- paste(sprintf("Contains a list of %d unique Test Short Name%s (%s) and Test Name%s (%s): ", nrow(unique_tests), ifelse(nrow(unique_tests) == 1, "", "s"), testcd_col, ifelse(nrow(unique_tests) == 1, "", "s"), test_col),
+      "#'   \\describe{",
+      paste(apply(unique_tests, 1, function(row) {
+        paste(sprintf("#'     \\item{%s}{%s}", row[[testcd_col]], row[[test_col]]))
+      }), collapse = "\n"),
+      "#'   }",
```
Comment (bundfussr) [inline data-raw/create_sdtms_data.R:133]: What about using a table instead of a description list?
Screenshot 2025-05-23 171821

Comment (Lina2689) [reply data-raw/create_sdtms_data.R:133]: Thanks @bundfussr We will look into this for table display.

File: data-raw/create_sdtms_data.R (hunk 1 above)
Comment (Fanny-Gautier) [reply data-raw/create_sdtms_data.R:125]: Sorted alphabetically as per Test codes

File: data-raw/create_sdtms_data.R (hunk 2 above)
Comment (Fanny-Gautier) [reply data-raw/create_sdtms_data.R]: Table created instead of the list

**Comment by Fanny-Gautier** [comment]
> I was also wondering if the Key features section of the script should also include a line talking about this new functionality?

Hi @manciniedoardo ,
Could you please confirm which script should be modified? Thanks

File: R/tu_onco_recist.R [hunk 3]
```diff
@@ -32,4 +32,9 @@
 #'
 #' @source Generated dataset.
 #' @author Stefan Bundfuss
+#' @details Contains a list of 1 unique test code and name:
+#'   \describe{
+#'     \item{TUMIDENT}{Tumore Identification}
```
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply R/tu_onco_recist.R]: Yes please

**Comment by manciniedoardo** [comment]
> > I was also wondering if the Key features section of the script should also include a line talking about this new functionality?
> 
> Hi @manciniedoardo , Could you please confirm which script should be modified? Thanks

I was referring to the section starting from line 7

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Looks good! noticed one other potential improvement.

**Review by Fanny-Gautier — APPROVED** [review]
All comments have been implemented

File: R/qs_ophtha.R [hunk 4]
```diff
@@ -35,4 +35,37 @@
 #'   }
 #'
 #' @source Constructed using qs from the {pharmaversesdtm} package
+#' @details Contains a set of 29 unique Test Short Names and Test Names: \tabular{ll}{
+#'   \strong{QSTESTCD} \tab \strong{QSTEST} \cr
+#'   VFQ1  \tab Your Overall Health Is                    \cr
+#'   VFQ10 \tab Difficulty Noticing Objects to Side       \cr
+#'   VFQ11 \tab Difficulty Seeing How People React        \cr
```
Comment (bundfussr) [inline R/qs_ophtha.R:42]: I wonder if these values are in line with CDISC controlled terminology. Maybe we should create an issue to check and update if necessary.

**Review by bundfussr — APPROVED** [review]
